### PR TITLE
21370 integrate fuel 300

### PR DESCRIPTION
**Pull request by theseion**
https://pharo.manuscript.com/f/cases/21370/Integrate-Fuel-3-0-0
https://pharo.fogbugz.com/f/cases/20456/FuelTests-package-should-be-renamed-to-Fuel-Tests

Replaces https://github.com/pharo-project/pharo/pull/886

### CODE REVIEW

**Comment by theseion** [comment]
New Fuel version 3.0.0. All packages have been renamed to 'Fuel-XXX'. FuelCommandLineHandler and FuelSystem-FileRegistry have been merged into the Fuel-Tools package.

This includes renaming of FuelTests to Fuel-Tests. See
https://github.com/pharo-project/pharo/pull/886 and
https://pharo.fogbugz.com/f/cases/20456/FuelTests-package-should-be-renamed-to-Fuel-Tests.
Tests from other packages have also been moved into sub packages of Fuel-Tests.

Fuel 3.0.0 is the first Github version. Future versions will only be available from Github.

**Comment by theseion** [comment]
Note: This change can't be simply tested by loading the changes into an existing image, as it removes all of the old Fuel packages.

**Comment by theseion** [comment]
Need to fix up BaselineOfIDE and BaselineOfMorphic.

**Comment by theseion** [comment]
Now the Baselines use the correct new package names but the packages can't be found...?? @estebanlm any ideas?
I've moved the different special Fuel packages (FuelCommandLineHandler, FuelSystem-FileRegistry etc.) into a package named Fuel-Tools. The log from the failed PR check says that Fuel-Tools can't be found.

**Comment by theseion** [comment]
Forget that. The package is actually named Fuel-Tools-Pharo...

**Comment by theseion** [comment]
Build was successful but unstable.

File: src/Fuel-Platform-Core/FLExtensionWrapper.class.st [hunk 1]
```diff
@@ -19,12 +19,10 @@ FLExtensionWrapper class >> fromPragma: aPragma [
 
 { #category : #private }
 FLExtensionWrapper >> compileInTarget [
-	| methodSelector source |
-	source := self method methodClass instanceSide
+	| source |
+	source := self method methodClass theNonMetaClass
```
Comment (jecisc) [inline src/Fuel-Platform-Core/FLExtensionWrapper.class.st]: @theseion I think #theNonMetaClass will be deprecated. Maybe  removing #methodSelector and using #instanceSide were changes not ported to Fuel repository?

**Review by jecisc — CHANGES_REQUESTED** [review]
Seems good except my comment.

File: src/Fuel-Tools-Pharo/FuelOutStackDebugAction.class.st [hunk 2]
```diff
@@ -95,7 +95,7 @@ FuelOutStackDebugAction >> serializeStack [
 	fileName := String streamContents: [ :s|
 		s 
 			<< 'Debugger-Stack-';
-			<< self interruptedContext receiver class instanceSide name;
+			<< self interruptedContext receiver class theNonMetaClass name;
```
Comment (jecisc) [inline src/Fuel-Tools-Pharo/FuelOutStackDebugAction.class.st]: Oh, I missed this one. It should be #instanceSide too I think. Probably a change not ported to Fuel repository.

Comment (theseion) [reply src/Fuel-Tools-Pharo/FuelOutStackDebugAction.class.st]: Thanks. I'll fix that too.

**Comment by guillep** [comment]
`testExternalMorphicDependencies` is failing.

**Comment by theseion** [comment]
Thanks Guille, I'll look into it.

**Comment by tinchodias** [comment]
I wasn't following this conversation. Very nice!

**Comment by theseion** [comment]
Now we should be good...

**Comment by stepharoself** [comment]
Thanks you guys! You are super cool :)

**Comment by guillep** [comment]
@theseion Can you explain where does it come from this new dependency to DebuggerActions?

It is strange that morphic depends on that (and not for example the opposite)...

**Comment by theseion** [comment]
I would also have expected the opposite.
`FuelOutStackDebugAction` was formerly part of the `DebuggerActions` package. I moved it to `Fuel-Tools`. I did the same thing for `FuelCommandLineHandler` which is now also part of `Fuel-Tools` and Morphic had a dependency on `FuelCommandLineHandler`. Hence, Morphic now has a dependency on `Fuel-Tools` and because `FuelOutStackDebugAction` is in that package and inherits from `DebugAction` Morphic now has an external dependency on `DebuggerActions`.

**Pull request merged**